plugins/catalog-backend/src/catalog/types.ts

```diff
@@ -41,9 +41,19 @@ export type EntitiesResponse = {
   pageInfo: PageInfo;
 };
 
+// TODO: Think a bit more about this type! We might want to split it into
+// metadata and the actual data. For reading we might only want to return the
+// metadata except the data itself is requested explicitly!
+export type EntityAttachment = {
+  key: string;
+  data: Buffer;
+  contentType: string;
+};
```

Comment:
My idea here is to have some kind of "conditional update". On every ingestion loop we output only the metadata from each attachment (`key` and `contentType`) and a callback which the API can use to retrieve the attachment. The callback takes an `etag` as an argument and returns an `etag` and the data. The backend calls the callback and skips updating the data in the database if the callback throws a `NotModifiedError`. However, I'm not sure if that is a good idea.